extension/app/ts/backgroundServiceWorker.ts

```diff
@@ -28,3 +28,36 @@ const injectContentScript = async () => {
 }
 
 injectContentScript()
+
+/// Workaround for ChromiumIssue1316588 https://bugs.chromium.org/p/chromium/issues/detail?id=1316588
+// TODO, remove when chrome fixes the bug
+const storageArea = chrome.storage.local as chrome.storage.LocalStorageArea & { onChanged: chrome.storage.StorageChangedEvent }
+
+const TEST_INTERVAL_MS = 10000
+const STORAGE_WAIT_TIME_MS = 100
+
+const hasChromiumIssue1316588 = () => {
+	return new Promise((resolve) => {
+		let dispatched = false
+		const testEventDispatching = () => {
+			storageArea.onChanged.removeListener(testEventDispatching)
+			dispatched = true
+		}
+		storageArea.onChanged.addListener(testEventDispatching)
+		storageArea.set({ testEventDispatching: Math.random() })
+		setTimeout(() => resolve(!dispatched), STORAGE_WAIT_TIME_MS)
+	})
+}
```

Comment:
We have an `async sleep(milliseconds: number)` helper function that makes it so we can use async functions rather than having to do this sort of manual promise management.
```suggestion
const hasChromiumIssue1316588 = async () => {
	let dispatched = false
	const testEventDispatching = () => {
		storageArea.onChanged.removeListener(testEventDispatching)
		dispatched = true
	}
	storageArea.onChanged.addListener(testEventDispatching)
	storageArea.set({ testEventDispatching: Math.random() })
	await sleep(STORAGE_WAIT_TIME_MS)
	return !dispatched
}
```